### ISSUE TITLE
Fix automatically generated PSA test cases that are not executed

### ISSUE THREAD

**Issue by gilles-peskine-arm** [Mbed-TLS/mbedtls#9169]
As part of https://github.com/Mbed-TLS/mbedtls/issues/5390, fix all or at least most instances of generated test cases that are not executed. This may include both arranging to execute tests that we're currently wrongly skipping, and arranging not to generate test cases that we can't execute.

If there are thorny cases that are hard to fix, it's ok to file a separate issue for them. The completeness goal here is an informal one: there should be few warnings so that we can decently read them one by one.

Prerequisites: https://github.com/Mbed-TLS/mbedtls/issues/9167, https://github.com/Mbed-TLS/mbedtls/issues/9168.
